File: Backend/Models/Entities/Application.cs
namespace InternAttendenceSystem.Models.Entities
{
    public class Application
    {


        public Guid applicationId { get; set; }
        public string FullName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string Gender { get; set; }
        public string ContactNumber { get; set; }
        public string Email { get; set; }
        public string Address { get; set; }
        public string Nationality { get; set; }

        public string  department { get; set; } 
        public string EducationLevel { get; set; }
        public string UniversityName { get; set; }
        public string Degree { get; set; }
        public string YearOfStudy { get; set; }
        public string EnrollmentNumber { get; set; }
        public double? CGPA { get; set; }


        public string InternshipRole { get; set; }
        public DateTime PreferredStartDate { get; set; }
        public DateTime PreferredEndDate { get; set; }
        public string ModeOfInternship { get; set; }
        public string LocationPreference { get; set; }


        public string TechnicalSkills { get; set; }
        public string SoftSkills { get; set; }
        public string PreviousExperience { get; set; }
        public string ResumeUrl { get; set; }
        public string ProfilePicUrl { get; set; }
        public string PortfolioUrl { get; set; }
        public string LinkedInUrl { get; set; }
        public string GitHubUrl { get; set; }

        // 🔹 Additional Information
        public string MotivationStatement { get; set; }
        public string SourceOfApplication { get; set; } // Referral, Website, etc.
        public string Availability { get; set; } // Full-time / Part-time
        public string EmergencyContact { get; set; }

        // 🔹 Compliance
        public bool AgreementAccepted { get; set; } // Terms and Conditions
        public DateTime RequestDate { get; set; } = DateTime.UtcNow; // Auto set
        public DateTime applicationDate { get; set; } // Added for recent activities

        public string Status { get; set; } 

    }
}

File: Frontend/src/app/models/application.ts
export interface ApplicationDto {
  fullName: string;
  dateOfBirth: Date;
  gender: string;
  contactNumber: string;
  email: string;
  address: string;
  nationality: string;
  educationLevel: string;
  universityName: string;
  degree: string;
  yearOfStudy: string;
  enrollmentNumber: string;
  cgpa?: number;
  internshipRole: string;
  preferredStartDate: Date;
  preferredEndDate: Date;
  modeOfInternship: string;
  locationPreference: string;
  technicalSkills: string;
  softSkills: string;
  previousExperience: string;
  resumeUrl: string;
  portfolioUrl: string;
  linkedInUrl: string;
  gitHubUrl: string;
  motivationStatement: string;
  sourceOfApplication: string;
  availability: string;
  emergencyContact: string;
  agreementAccepted: boolean;
  requestDate: Date;
}

export interface ApplicationResponseDto {
  applicationId: string;
  fullName: string;
  dateOfBirth: Date;
  gender: string;
  contactNumber: string;
  email: string;
  address: string;
  nationality: string;
  status: string; // e.g., Pending, Reviewed, Accepted, Rejected
  educationLevel: string;
  universityName: string;
  degree: string;
  yearOfStudy: string;
  enrollmentNumber: string;
  cgpa?: number;
  internshipRole: string;
  preferredStartDate: Date;
  preferredEndDate: Date;
  modeOfInternship: string;
  locationPreference: string;
  technicalSkills: string;
  softSkills: string;
  previousExperience: string;
  resumeUrl: string;
  profilePicUrl: string;
  portfolioUrl: string;
  linkedInUrl: string;
  gitHubUrl: string;
  motivationStatement: string;
  sourceOfApplication: string;
  availability: string;
  emergencyContact: string;
  agreementAccepted: boolean;
  requestDate: Date;
}

export interface CreateApplicationDto {
  fullName: string;
  dateOfBirth: Date;
  gender: string;
  contactNumber: string;
  email: string;
  address: string;
  nationality: string;
  educationLevel: string;
  department: string; // Added this line
  universityName: string;
  degree: string;
  yearOfStudy: string;
  enrollmentNumber: string;
  cgpa?: number;
  internshipRole: string;
  preferredStartDate: Date;
  preferredEndDate: Date;
  modeOfInternship: string;
  locationPreference: string;
  technicalSkills: string;
  softSkills: string;
  previousExperience: string;
  resumeFile?: File; // For file upload
  profilePicFile?: File; // For file upload
  portfolioUrl: string;
  linkedInUrl: string;
  gitHubUrl: string;
  motivationStatement: string;
  sourceOfApplication: string;
  availability: string;
  emergencyContact: string;
  agreementAccepted: boolean;
}

export interface UpdateApplicationDto {
  fullName?: string;
  dateOfBirth?: Date;
  gender?: string;
  contactNumber?: string;
  email?: string;
  status?: string; // e.g., Pending, Reviewed, Accepted, Rejected
  address?: string;
  nationality?: string;
  educationLevel?: string;
  universityName?: string;
  degree?: string;
  yearOfStudy?: string;
  enrollmentNumber?: string;
  cgpa?: number;
  internshipRole?: string;
  preferredStartDate?: Date;
  preferredEndDate?: Date;
  modeOfInternship?: string;
  locationPreference?: string;
  technicalSkills?: string;
  softSkills?: string;
  previousExperience?: string;
  resumeUrl?: string;
  portfolioUrl?: string;
  linkedInUrl?: string;
  gitHubUrl?: string;
  motivationStatement?: string;
  sourceOfApplication?: string;
  availability?: string;
  emergencyContact?: string;
  agreementAccepted?: boolean;
}

export interface UpdateApplicationStatusDto {
  status: string;
}
